Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs

```diff
@@ -4805,169 +5124,34 @@ private void UpdateCylinders(float elapsedClockSeconds, float throttle, float cu
-            CombFrictionN = 0;
-            CombGravityN = 0;
-            CombTunnelN = 0;
-            CombCurveN = 0;
-            CombWindN = 0;
-
-
-            LocoIndex = 0;
-            for (int i = 0; i < Train.Cars.Count; i++)  // Doesn't included the locomotive or tender
-            {
-                if (Train.Cars[i] == this)
-                    LocoIndex = i;
-            }
-
-            // Find frictional forces of the locomotive
-            CombFrictionN = Train.Cars[LocoIndex].FrictionForceN;
-            CombGravityN -= Train.Cars[LocoIndex].GravityForceN;
-            CombTunnelN = Train.Cars[LocoIndex].TunnelForceN;
-            CombCurveN = Train.Cars[LocoIndex].CurveForceN;
-            CombWindN = Train.Cars[LocoIndex].WindForceN;
-
-            if (HasTenderCoupled)
-            {
-                if (LocoIndex < Train.Cars.Count - 1) // Room for a tender in the train
-                {
-                    // Find frictional forces of tender and add them to the locomotive
-                    CombFrictionN += Train.Cars[LocoIndex + 1].FrictionForceN;
-                    CombGravityN -= Train.Cars[LocoIndex + 1].GravityForceN; // Gravity forces have negative values on rising grade
-                    CombTunnelN += Train.Cars[LocoIndex + 1].TunnelForceN;
-                    CombCurveN += Train.Cars[LocoIndex + 1].CurveForceN;
-                    CombWindN += Train.Cars[LocoIndex + 1].WindForceN;
-                }
-
+                CalculatedCylinderSteamUsageLBpS = 0.0001f; // Set steam usage to a small value if throttle is closed
             }
 
-            LocoTenderFrictionForceN = CombFrictionN + CombGravityN + CombTunnelN + CombCurveN + CombWindN;  // Combined frictional forces of the locomotive and tender
+            // usage calculated as moving average to minimize chance of oscillation.
+            // Decrease steam usage by SuperheaterUsage factor to model superheater - very crude model - to be improved upon
+            SteamEngines[numberofengine].CylinderSteamUsageLBpS = (0.6f * SteamEngines[numberofengine].CylinderSteamUsageLBpS + 0.4f * CalculatedCylinderSteamUsageLBpS);
+
+            MeanEffectivePressurePSI = SteamEngines[numberofengine].MeanEffectivePressurePSI; // for display purposes
+
+            SteamReleasePressure_AtmPSI = SteamEngines[numberofengine].Pressure_c_AtmPSI; // for steam and smoke effects
 
-            MotiveForceSmoothedN.Update(elapsedClockSeconds, MotiveForceN);
-            if (float.IsNaN(MotiveForceN))
-                MotiveForceN = 0;
         }
 
-        protected override void UpdateTractiveForce(float elapsedClockSeconds, float t, float AbsSpeedMpS, float AbsWheelSpeedMpS)
+        private void UpdateMotion(float elapsedClockSeconds, float cutoff, float absSpeedMpS)
+        {
+
+
+        }
+
+        protected override void UpdateTractiveForce(float elapsedClockSeconds, float t, float AbsSpeedMpS, float AbsWheelSpeedMpS, int numberofengine)
```

Comment:
Again, can we move this to SteamEngine.UpdateTractiveForce(), and then here we just do a foreach loop to call it once for each engine?